packages/subproviders/src/subproviders/base_wallet_subprovider.ts

```diff
@@ -0,0 +1,134 @@
+            case 'eth_coinbase':
+                try {
+                    accounts = await this.getAccountsAsync();
+                    end(null, accounts[0]);
+                } catch (err) {
+                    end(err);
+                }
+                return;
+
+            case 'eth_accounts':
+                try {
+                    accounts = await this.getAccountsAsync();
+                    end(null, accounts);
+                } catch (err) {
+                    end(err);
+                }
+                return;
+
+            case 'eth_sendTransaction':
+                txParams = payload.params[0];
+                try {
+                    BaseWalletSubprovider._validateSender(txParams.from);
+                    const filledParams = await this._populateMissingTxParamsAsync(txParams);
+                    const signedTx = await this.signTransactionAsync(filledParams);
+                    const response = await this._emitSendTransactionAsync(signedTx);
+                    end(null, response.result);
+                } catch (err) {
+                    end(err);
+                }
+                return;
+
+            case 'eth_signTransaction':
+                txParams = payload.params[0];
+                try {
+                    const filledParams = await this._populateMissingTxParamsAsync(txParams);
+                    const signedTx = await this.signTransactionAsync(filledParams);
+                    const result = {
+                        raw: signedTx,
+                        tx: txParams,
+                    };
+                    end(null, result);
+                } catch (err) {
+                    end(err);
+                }
+                return;
+
+            case 'eth_sign':
+            case 'personal_sign':
+                const data = payload.method === 'eth_sign' ? payload.params[1] : payload.params[0];
+                try {
+                    const ecSignatureHex = await this.signPersonalMessageAsync(data);
+                    end(null, ecSignatureHex);
+                } catch (err) {
+                    end(err);
+                }
+                return;
+
+            default:
+                next();
+                return;
+        }
+    }
+    private async _emitSendTransactionAsync(signedTx: string): Promise<JSONRPCResponsePayload> {
+        const payload = {
+            method: 'eth_sendRawTransaction',
+            params: [signedTx],
+        };
+        const result = await this.emitPayloadAsync(payload);
+        return result;
+    }
+    private async _populateMissingTxParamsAsync(txParams: PartialTxParams): Promise<PartialTxParams> {
+        if (_.isUndefined(txParams.gasPrice)) {
+            const gasPriceResult = await this.emitPayloadAsync({
+                method: 'eth_gasPrice',
+                params: [],
+            });
+            const gasPrice = gasPriceResult.result.toString();
+            txParams.gasPrice = gasPrice;
```

Comment:
we're mutating the input here, could be a source of bugs in the future

Reply:
As in you think we should spread and create a new params object? That should be fine as we always re-assign rather than assume its modified in place